src/message_header.rs

```diff
@@ -156,11 +157,12 @@ impl MessageHeader {
         reply_header.source = types::SourceAddress {
             from_node : our_name.clone(),
             from_group : Some(self.destination.dest.clone()),
-            reply_to : None
+            reply_to : None,
+            relayed_for: None
         };
         reply_header.destination = types::DestinationAddress {
             dest : self.from().clone(),
```

Comment:
This was wrong in the old code: can we make
dest : match self.source.reply_to {
   Some(reply_to) => reply_to,
   None => match self.from_group {
      Some(group_name) => group_name,
      None => self.from_node